docs/lang/sync.md

```diff
@@ -5,83 +5,52 @@ synchronization in presence of parallelism.
 We're currently experimenting with a suite of new primitives that add synchronization to the
 language.
 
-## `std_sync_reg`
+## `@sync` attribute
 
-The `std_sync_reg` primitive defined by `primitives/sync.futil` provides a synchronizing
-register that acts as an [M-structure][m-struct] which provides the following interface:
-On the reader side:
-1. If the register is "empty", block the read till the register is written into.
-2. If the register is "full", provide the value to the reader, provide a `done` signal, and mark it as "empty".
+## Motivation
 
-On the writer side:
-1. If the register is "empty", write the value in the register, mark it as "full", and provide a `done` signal.
-2. If the register is "full", block the write till the register is read from.
+Consider the following control program in calyx:
+```
+{{#include ../../examples/sync/unsync-example/unsync-doc-example.futil:control}}
+```
+
+where groups `add_r_to_accm` and `incr_r` reads value and increments value in register `r`, respectively, as indicated by their names. 
 
-One way to think of this interface is as a size-1 concurrent FIFO.
+Because calyx does not make any guarantee of the order of execution for threads running in parallel, it is impossible for us to determine which thread will access r first for each iteration.
 
-## Using `std_sync_reg`
+Nondeterminism when running parallel threads is beneficial on the compiler's end, as it will give the compiler more freedom for optimization. However, we sometimes do want to give parallel threads a measure of ordering while still taking advantage of the performance boost of parallelism. The @sync attribute allows us to do that. 
 
-> The [following example][ex] is a part of the Calyx compiler test suite and can be
-> executed using:
->
->       runt -i examples/sync
 
-The synchronizing register interface is non-standard: it provides two go signals and
-two done signals to initiate parallel reads and writes.
+## Using the `@sync` attribute
+
+Now we want to modify the program above so that in every iteration, thread A always reads after thread B finishes incrementing using the `@sync` attribute with the following:
 
 ```
-{{#include ../../primitives/sync.futil:sync_interface}}
+{{#include ../../examples/sync/sync-doc-example.futil:control}}
```

Comment:
Why does this program have two `@sync` annotations? I would've expected it only needs one

Reply:
Because if we only have one @sync, then the second time incr_r is executed might happen before the first time add_r_to_accm is executed.

Reply:
Got it, that makes sense!